lntest/itest/lnd_send_payment_multi_path.go

```diff
@@ -0,0 +1,271 @@
+// +build rpctest
+
+package itest
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/btcsuite/btcd/wire"
+	"github.com/btcsuite/btcutil"
+	"github.com/lightningnetwork/lnd"
+	"github.com/lightningnetwork/lnd/lnrpc"
+	"github.com/lightningnetwork/lnd/lnrpc/routerrpc"
+	"github.com/lightningnetwork/lnd/lntest"
+)
+
+// testSendPaymentMultiPath tests that we are able to successfully route a
+// payment using multiple shards across different paths.
+func testSendPaymentMultiPath(net *lntest.NetworkHarness, t *harnessTest) {
+	ctxb := context.Background()
+
+	fmt.Printf("DEBUG %v: test start\n", time.Now())
+
+	// To ensure the payment goes through seperate paths, we'll set a
+	// channel size that can only carry one shard at a time. We'll divide
+	// the payment into 3 shards. We'll also set a small push amt, to make
+	// sure we don't violate channel reserves when we send a payment back.
+	const paymentAmt = btcutil.Amount(300000)
+
+	// Set up a network with three different paths Alice <-> Bob.
+	//              _ Eve _
+	//             /       \
+	// Alice -- Carol ---- Bob
+	//      \              /
+	//       \__ Dave ____/
+	nodes := make(map[string]*lntest.HarnessNode)
+	nodes["alice"] = net.Alice
```

Comment:
avoid using alice and bob?

Reply:
Why is that? Because of state carry-over? I now reuse the test context that is used for the multi-sendtoroute test